fatal/type/impl/type.h

```diff
@@ -15,18 +15,20 @@
 namespace fatal {
 namespace impl_t {
 
+template <typename T, typename = typename T::type>
+std::true_type s(T *);
+std::false_type s(...);
+
+template <typename T, bool = decltype(s(static_cast<T *>(nullptr)))::value>
```

Comment:
Looks like we need a `void_t`, so that this could look like:

```
template <typename T, typename = void_t<typename T::type>>
```
https://github.com/facebook/folly/blob/v2017.08.28.00/folly/Traits.h#L215-L223